### PR TITLE
create modal context

### PR DESCRIPTION
**Pull request by jgfidelis**

### CODE REVIEW

File: react/ModalContext.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React, { createContext } from 'react'
+
+const { Consumer, Provider } = createContext({})
+
+interface ModalContext {
+  closeModal: () => void
+}
+
+const modalConsumer = <P extends object>(
```
Comment (lucis) [inline react/ModalContext.tsx]:
```suggestion
const withModal = <P extends object>(
``` 
may be more consistent

File: react/ModalContext.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React, { createContext } from 'react'
+
+const { Consumer, Provider } = createContext({})
+
+interface ModalContext {
+  closeModal: () => void
```
Comment (lucis) [inline react/ModalContext.tsx:6]: No `openModal`?

File: react/ModalContext.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+const modalConsumer = <P extends object>(
+    WrappedComponent: React.ComponentType<P>
+  ): React.FC<P & ModalContext> => 
+  (props: P) => (
+    <Consumer>
+      {context => <WrappedComponent {...props} modalContext={context} />}
```
Comment (lucis) [inline react/ModalContext.tsx]:
```suggestion
      {context => <WrappedComponent {...props} closeModal={context.closeModal} />}
```

**Review by lucis — APPROVED** [review]

File: react/ModalContext.tsx (hunk 2 above)
Comment (lbebber) [reply react/ModalContext.tsx:6]: @luciannojunior I think this context will only be applied to children of the modal, so from their point of view, the modal has always been open (I might be mistaken though)

Comment (jgfidelis) [reply react/ModalContext.tsx:6]: I agree with @lbebber and I don't think openModal should be included, at least for now.

**Pull request merged**